app/frontend/lectures/lectures.js

```diff
@@ -353,6 +335,28 @@ $(document).on("turbo:load", function () {
   });
 });
 
+/**
+ * Handles updating the "Create New Medium" button's href
+ * to include the current sort parameter when switching tabs.
+ *
+ * This is such that when a user clicks the "Create New Medium" button,
+ * the new medium will be created with the same sort as the currently active tab.
+ */
+function handleNewMediaButtonSort() {
+  $('#edited-media-tab a[data-bs-toggle="tab"]').on("show.bs.tab", function (e) {
+    const button = $("#create-new-medium");
+    const sort = e.target.dataset.sort;
+    if (!sort) {
+      console.error("No sort type found on tab!");
+      return;
+    }
+    const currentHref = button.prop("href");
+    const url = new URL(currentHref, window.location.origin);
+    url.searchParams.set("sort", sort);
+    button.prop("href", url.pathname + url.search);
+  });
+}
+
```

Comment:
Have you considered extracting this into a stimulus controller? It seems like a viable candidate and could offer a clean solution that also thins out this already rather large lectures.js file (also, everything can be done with nice data attributes instead of element ids).